pkg/component/component.go

```diff
@@ -76,6 +77,60 @@ type Unit struct {
 	Err error `yaml:"error,omitempty"`
 }
 
+type Signed struct {
+	Data      string `yaml:"data"`
+	Signature string `yaml:"signature"`
+}
+
+// Checks if the signature exists, safe to call on nil
+func (s *Signed) IsSigned() bool {
+	return (s != nil && (len(s.Signature) > 0))
+}
+
+var ErrNotFound = errors.New("not found")
+
+func SignedFromPolicy(policy map[string]interface{}) (*Signed, error) {
+	v, ok := policy["signed"]
+	if !ok {
+		return nil, fmt.Errorf("policy is not signed: %w", ErrNotFound)
```

Comment:
Should this maybe be a defined error? `ErrNotSigned`?

Reply:
from general use case point of view, it's the same meaning, we care if we have data or not, same for the tests.
The ```fmt.Errorf``` adds additional context to ```ErrNotFound```, which can be logged or propagated up. 
I might be more convenient if we could wrap multi error probably like it's possible with Go 1.20
https://tip.golang.org/doc/go1.20#errors